src/azure-cli/azure/cli/command_modules/acs/_validators.py

```diff
@@ -73,6 +75,22 @@ def validate_create_parameters(namespace):
         raise CLIError('--dns-prefix has no value')
 
 
+def validate_ip_ranges(namespace):
+    if not namespace.api_server_authorized_ip_ranges:
+        return
+
+    allow_all_traffic = "0.0.0.0/32"
+    for ip in namespace.api_server_authorized_ip_ranges.split(','):
+        if ip == allow_all_traffic:
+            continue
+        try:
+            ip = ip_network(ip)
+            if not ip.is_global:
+                raise CLIError("--api-server-authorized-ip-ranges must be global non-reserved addresses or CIDRs")
+        except ValueError:
```

Comment:
Just confirming since not so good with python. Empty string "" will not throw this error right?

Reply:
It would but I had originally thought the for loop that splits the string into an array would not split "" into an array of any values -- thereby not entering the for loop. I re-tested that and I was wrong. 

I have set a few new safety checks to guard against a customer setting "" and some other IPs (which is illogical) and also to skip the IP global check for empty string. 

Thanks for being so observant :)